### PR TITLE
Don't throw if we're called on a background thread.  

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Just return an empty set of tags.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Reviewers @balajikris, @basoundr, @brettfo, @davkean, @DustinCampbell, @dpoeschl, @Pilchie @jasonmalinowski

File: src/ExpressionEvaluator/Core/Source/ResultProvider/NetFX20/ResultProvider.NetFX20.csproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <TargetNetFX20>true</TargetNetFX20>
+    <NuGetTargetMoniker>.NETFramework,Version=v2.0</NuGetTargetMoniker>
```
Comment (CyrusNajmabadi) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/NetFX20/ResultProvider.NetFX20.csproj]: Unrelated.  These are from another branch that i had to branch off of.  It's the only way i could build/test.  Once my other PR goes in and is merged, these will disappear from this PR.

**Comment by jasonmalinowski** [comment]
Why? Do we understand who is calling us on the background thread to know if this is safe?

**Comment by CyrusNajmabadi** [comment]
There is an extension out there that is calling us on a background thread.  All this does is prevent us from crashing, and we instead just give them an empty list.  I imagine an empty list is 'safer' than crashing :)

**Comment by CyrusNajmabadi** [comment]
I've followed up with the platform team on this.  It seems like the right thing to do is to keep our current behavior, but reach out to the Extension author to fix their code to not do things that expect thread-safety where none is guaranteed.

The platform team is going to be driving the extension experience, and they want to be able to provide a better experience here.  i.e. one where the user is made aware of the issue with the extension so they can potentially disable it to get back into a better state.  

So i'm closing this for now.

**Pull request closed without merging**